### PR TITLE
Prow job to build docker and containerd packages

### PR DESCRIPTION
**Pull request by florencepascual**
This prow job is a periodic prow job, that should spawn a docker in docker environment.

### CODE REVIEW

**Comment by mkumatag** [comment]
@florencepascual what is the goal here? what are we trying to achieve here?

**Comment by florencepascual** [comment]
@mkumatag Our goal here is to get a docker in docker environment for testing our future docker builds

**Comment by mkumatag** [comment]
> @mkumatag Our goal here is to get a docker in docker environment for testing our future docker builds

okay.

**Comment by mkumatag** [comment]
> @mkumatag Our goal here is to get a docker in docker environment for testing our future docker builds

at the moment we don't have any PR run for validating this job before merging it, you may need to run this PR offline and confirm on a ppc64le to see if really works or not!

File: config/jobs/periodic/docker-in-docker/periodic-all-in-one.yaml
Comment (clnperez) [inline config/jobs/periodic/docker-in-docker/periodic-all-in-one.yaml:1]: instead of a periodic job, can this be a presubmit job that runs against any 20-* branch in docker-ce-packaging?

Kind of the idea but probably not quite right (and still has some bit from https://github.com/kubernetes/test-infra/blob/master/prow/jobs.md):

presubmits:
  docker/docker-ce-packaging:
  - name: docker-all-in-one-test
    decorate: true 
    always_run: true   
    skip_report: true  
    max_concurrency: 5
    spec: {your spec...}
    branches:             # Regexps, only run against these branches.
    - ^20
    trigger: "(?m)qux test this( please)?" # Regexp, see discussion.
    rerun_command: "qux test this please"  # String, see discussion.

File: config/jobs/periodic/docker-in-docker/periodic-all-in-one.yaml
Comment (clnperez) [inline config/jobs/periodic/docker-in-docker/periodic-all-in-one.yaml:10]: remind me again why we're not using https://hub.docker.com/_/docker? did it not exist when we started this b/c we didn't have docker official support for power at the time?

**Review by clnperez — COMMENTED** [review]
one request about changing to a `presubmit job`, and also would like a reminder about the all-in-on-dind image history

File: config/jobs/periodic/docker-in-docker/periodic-all-in-one.yaml
Comment (mkumatag) [reply config/jobs/periodic/docker-in-docker/periodic-all-in-one.yaml:1]: In order to do this, we need to have webhook registered into that repo and some write access to the repository to reflect the test results etc.. which doesn't look like a feasible idea atm, having that said we are working on a plan to run the presubmit/postsubmit jobs for the external owner repos,.. will keep you posted!

**Comment by Rajalakshmi-Girish** [comment]
> 
> 
> > @mkumatag Our goal here is to get a docker in docker environment for testing our future docker builds
> 
> at the moment we don't have any PR run for validating this job before merging it, you may need to run this PR offline and confirm on a ppc64le to see if really works or not!

Shall I run this on my workspace and confirm if it works on ppc?
Hope we needn't run against the actual ppc64le build cluster that our prow CI uses.

File: config/jobs/periodic/docker-in-docker/periodic-all-in-one.yaml
Comment (florencepascual) [reply config/jobs/periodic/docker-in-docker/periodic-all-in-one.yaml:1]: Can I ask you more details about that plan ?

**Comment by florencepascual** [comment]
> one request about changing to a `presubmit job`, and also would like a reminder about the all-in-on-dind image history

We chose to use this customized image https://quay.io/repository/alunsin/all_in_one_dind, since there was at the moment no support for ppc64le. If I am not mistaken, there is still no official support and no working docker-in-docker images for ppc64le.

File: config/jobs/periodic/docker-in-docker/periodic-all-in-one.yaml
Comment (mkumatag) [reply config/jobs/periodic/docker-in-docker/periodic-all-in-one.yaml:1]: It may take some more time as we are busy with other deliverables, keep one thing in mind that this may be additional permissions like PR status write access, PR comment access etc, so even it also depends on whether the project is comfortable giving those access to the bot.

Comment (clnperez) [reply config/jobs/periodic/docker-in-docker/periodic-all-in-one.yaml:1]: > we are working on a plan to run the presubmit/postsubmit jobs for the external owned repos

which repos? and who's `we`? the runtimes squad has been working with docker for over a year now so we definitely should stay in sync and not have two different groups of people working on this without communication at least. there's also the issue of the contract, etc.,

Comment (mkumatag) [reply config/jobs/periodic/docker-in-docker/periodic-all-in-one.yaml:1]: > > we are working on a plan to run the presubmit/postsubmit jobs for the external owned repos
> 
> which repos? and who's `we`? the runtimes squad has been working with docker for over a year now so we definitely should stay in sync and not have two different groups of people working on this without communication at least. there's also the issue of the contract, etc., 

Nothing to do with docker, we are just working on building a solution to run the jobs external owned repository, nothing to do with any individual project.. it is up to different squads if they want to use that solution...

Comment (clnperez) [reply config/jobs/periodic/docker-in-docker/periodic-all-in-one.yaml:1]: ok great. and thanks. just an english-ism that sounded like you had specific repos targeted.

**Review by Rajalakshmi-Girish — CHANGES_REQUESTED** [review]
I tried running this job yaml against my ppc64le k8s cluster. The job got struck for 2hs and timed out after that.
The run is available at https://prow.ppc64le-cloud.org/view/s3/prow-logs/logs/docker-all-in-one-test/1427888024015867904
Ran the test-pj as mentioned in https://github.com/ppc64le-cloud/test-infra#method-2-test-on-the-already-deployed-kubernetes-cluster
Here is the pod log which is available in the above link as well.
[dind_pod.log](https://github.com/ppc64le-cloud/test-infra/files/7006321/dind_pod.log)
Can you please take a look at it and confirm if I missed something.
Also, I have a minor suggestion of changing the job yaml file name periodic-all-in-one.yaml to something else mentioning dind in the name.

**Comment by Rajalakshmi-Girish** [comment]
@florencepascual Here is the successful run https://prow.ppc64le-cloud.org/view/s3/prow-logs/logs/docker-all-in-one-test/1428339709515403264. Hope it was as expected.

**Comment by florencepascual** [comment]
> @florencepascual Here is the successful run https://prow.ppc64le-cloud.org/view/s3/prow-logs/logs/docker-all-in-one-test/1428339709515403264. Hope it was as expected.

Thank you! Yes the log is as it is supposed to be.

**Comment by florencepascual** [comment]
This is a periodic prow job, which would build the docker and containerd packages for ppc64le in a semi-automated manner.

**Comment by florencepascual** [comment]
This PR was a first test in a docker in docker environment.
The following PR, https://github.com/ppc64le-cloud/test-infra/pull/233 is the next step with the docker and containerd build in a semi-automated manner.

**Comment by florencepascual** [comment]
See https://github.com/ppc64le-cloud/test-infra/pull/234 (squashed commits problem)

**Pull request closed without merging**
